tdrs-backend/tdpservice/parsers/schema_defs/tanf/t7.py

```diff
@@ -15,45 +15,83 @@
 families_index = 10
 for i in range(1, 31):
     month_index = (i - 1) % 3
-    sub_item_labels = ['A', 'B', 'C']
+    sub_item_labels = ["A", "B", "C"]
     families_value_item_number = f"6{sub_item_labels[month_index]}"
 
     schemas.append(
-      RowSchema(
-          model=TANF_T7,
-          quiet_preparser_errors=i > 1,
-          preparsing_validators=[
-              validators.notEmpty(0, 7),
-              validators.notEmpty(validator_index, validator_index + 24),
-          ],
-          postparsing_validators=[],
-          fields=[
-              Field(item="0", name="RecordType", type='string', startIndex=0, endIndex=2,
-                    required=True, validators=[]),
-              Field(item="3", name='CALENDAR_QUARTER', type='number', startIndex=2, endIndex=7,
-                    required=True, validators=[validators.dateYearIsLargerThan(1998),
-                                               validators.quarterIsValid()]),
-              TransformField(
-                  transform_func=calendar_quarter_to_rpt_month_year(month_index),
-                  item="3A",
-                  name='RPT_MONTH_YEAR',
-                  type='number',
-                  startIndex=2,
-                  endIndex=7,
-                  required=True,
-                  validators=[
-                      validators.dateYearIsLargerThan(1998),
-                      validators.dateMonthIsValid()
-                  ]
-              ),
-              Field(item="4", name='TDRS_SECTION_IND', type='string', startIndex=section_ind_index,
-                    endIndex=section_ind_index + 1, required=True, validators=[validators.oneOf(['1', '2'])]),
-              Field(item="5", name='STRATUM', type='string', startIndex=stratum_index,
-                    endIndex=stratum_index + 2, required=True, validators=[validators.isInStringRange(1, 99)]),
-              Field(item=families_value_item_number, name='FAMILIES_MONTH', type='number', startIndex=families_index,
-                    endIndex=families_index + 7, required=True, validators=[validators.isInLimits(0, 9999999)]),
-          ]
-      )
+        RowSchema(
+            model=TANF_T7,
+            quiet_preparser_errors=i > 1,
+            preparsing_validators=[
+                validators.hasLength(247),
```

Comment:
thanks for adding the row length check @elipe17. now when the record is too short, we get the proper feedback in the error report, the record isnt parsed and added to the db, but the status seems incorrect. shouldnt we have a status of "rejected" when the M7 record in the file cant be parsed? 
![S4pr](https://github.com/raft-tech/TANF-app/assets/63075587/fb1d4738-55c2-4cca-bb4d-7595aa7b222d)

Reply:
#2754